### DISCUSSION TITLE
I can't change my repo URL or copy my workflows

### DISCUSSION THREAD

**Discussion by thassio-vinicius** [codemagic-ci-cd/codemagic-docs#2184] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I would like to either change my repository URL or copy all my workflows to a new app. I created a v2 of the current app and made a new github repo with all the foundations and configurations, now I just need to integrate Codemagic with this new repo, but I have multiple workflows and it would be too tedious to copy everything manually. Since there would be no changes to the workflows, either copying the workflows or updating the current URL would work fine.

App URL: https://codemagic.io/app/65431169e3d83e9274601722
The current URL: https://github.com/TrezureTeam/mobile-app
The new URL: https://github.com/TrezureTeam/trezureV2

I couldn't find a place to copy my workflows (created with the Flutter workflow editor) and for some reason, the URL update in settings is readonly. Could somebody help me with this?

### Expected results

being able to update URL or copy workflows.

### Actual results

URL is readonly and there's no place to copy workflows

**Comment by dtrdic** [comment]
hey @thassio-vinicius, i'll change the URL for you but first you will need to provide me the new app id. In order to get a new app id, you will need to add your app ( https://github.com/TrezureTeam/trezureV2 ) to Codemagic first, then send me its id.

**Reply by Mirshodbek** [reply]
Hi, I seem to have a problem that it can't find repositories. But we haven't changed the repository link at all. A couple of days ago I built the build without any problems, but suddenly it stopped building. And then there's this problem. Can you please tell me how to solve this problem? Thanks

<img width="1728" alt="Снимок экрана 2025-01-25 в 10 34 48" src="https://github.com/user-attachments/assets/902641a7-2bd7-47ac-98c6-96ece9b00213" />

Cannot access the repository Not Found (404), the requested resource could not be found https://gitlab.com/api/v4/projects/50760924/repository/branches?per_page=100

<img width="1728" alt="Снимок экрана 2025-01-25 в 10 35 02" src="https://github.com/user-attachments/assets/834e7080-2964-481f-886a-2fe27c8a70f4" />
Cannot update repository https://gitlab.com/api/v4/projects/50760924 Repository was not found

**Comment by thassio-vinicius** [comment]
@dtrdic small update: the team had a last minute change to a new repo URL to comply with the current naming conventions, the new URL is: https://github.com/TrezureTeam/mobile-app-v2

And the codemagic app ID for this URL is: https://codemagic.io/app/65530a3ff369ce189f128751

Sorry for the confusion.

**Reply by dtrdic** [reply, marked as answer]
hey @thassio-vinicius i made the migration, can you please run a test build on a new app to check if it's all good?

**Reply by thassio-vinicius** [reply]
@dtrdic if I start a manual build directly from Codemagic it works, but the build triggers aren't working. The triggers were working fine using tag creation events before. I tried a few times but no luck, and I don't see anything in the webhooks tab either.

**Reply by dtrdic** [reply]
you added a new app, which means a new webhook as well. Just copy this new webhook to your git repository and everything should be fine. Check out [this doc](https://docs.codemagic.io/yaml-running-builds/webhooks/)

**Reply by thassio-vinicius** [reply]
It's working, thanks!

**Comment by omar-chaabouni** [comment]
@dtrdic Hello David, I have the same issue, I am trying to change my repo URL (used to be bitbucket integration), now I want it to be using "Other" integration. What should I do ?

**Reply by dtrdic** [reply]
Hey @omar-chaabouni, now you need to re-add this application to Codemagic (the one that is currently in a different repository). This way, you will have two identical applications inside Codemagic, but with different URL paths to the repositories. After that, share with me the App IDs of both the 'old app' and the 'new app,' and I will use those App IDs to perform the necessary migration

**Reply by omar-chaabouni** [reply]
Thank you for your quick response David @dtrdic 
Old project ID: 64b5a82e6d17f032632853d9
New project ID: 659d5b94583d005ba6ee4a0f
I don't know how it works exactly from your side.
Anyways, I think either you change the old project url to match the new project or coy the exact same configuration of the old project into the new one.

**Reply by dtrdic** [reply]
Updated your old app url with the URL of the new app. Please try to run a test build, let me know how it goes

**Reply by omar-chaabouni** [reply]
Got signing errors (my bad apparently), but your changes are successful. Thanks David 💯

**Comment by Moh-Afifi-1024** [comment]
Hey @dtrdic  I have the same issue and I need your support please.
I have two applications and I need to migrate from bitbucket to github
First App:
Old App Id: 63dad5358d8ae43d9774f9b1
New App Id: 67f29b71ffc2812baefa7837

Second App:
Old App Id: 63e150db94a6e72db44e1892
New App Id: 67f29b8b92bbe035290eebde

**Reply by dtrdic** [reply]
Hi @Moh-Afifi-1024 ,done!

**Reply by Moh-Afifi-1024** [reply]
Thanks a lot for your reply and efforts. It works !!

**Comment by srdjan86** [comment]
Hey @dtrdic I have the same issue and I need your support please.

Old App Id: 62e01c67097606d89bd65351
New App Id: 68304251ce1f7a472a3c76a3

**Reply by dtrdic** [reply]
@srdjan86 done!

**Reply by srdjan86** [reply]
Thank you @dtrdic for quick reply. I still see only one default workflow and am asked to complete setup

**Reply by dtrdic** [reply]
@srdjan86 what exactly is your issue?
The above threads were asking for changing the URL path of their repository and i did that the same for you, i canahged the URL of your old app from https://bitbucket.org/certaintysoftware/certainty-software --> https://bitbucket.org/certaintysoftware/certainty-app-v2

**Reply by srdjan86** [reply]
@dtrdic ooooh, perfect, right, thank you!
I was under impression that it is not possible to change the url of repository in application and that new application should be created (and probably used later).
Gotcha, thank you again, my original application now has updated repo url.
Many thanks again

**Comment by kevingaulin** [comment]
hey @dtrdic ! i need to migrate from a custom repo (devops) to github

old app:   6844ff644a0a25cbff94e8e0
new app: 687fe404e8f4dad59dff0aa1

**Reply by dtrdic** [reply]
hi @kevingaulin, done!

**Comment by isprodmctainsh** [comment]
Hi @dtrdic - Can you please help me migrate our bitbucket app to github?

Old app: 64e459a01d32960ee5345c1d (bitbucket)
New app: 6892b906bfc8e170e582cfbc (github)

Thank you!

**Reply by dtrdic** [reply]
Hi @isprodmctainsh, done

**Reply by isprodmctainsh** [reply]
Thank you very much!

**Comment by Adeiza-Dev** [comment]
Hi, I moved our repo to another org and I am having isssues updating the url on codemagic. How do I go about this. I keep getting 
Cannot access repositories
Not Found (404), the requested resource could not be found https://api.github.com/user/installations/35574968/repositories?sort=pushed&per_page=100

@dtrdic

**Reply by dtrdic** [reply]
Hi @Adeiza-Dev, the repo url cannot be changed through the UI. That's why you’ll need to add two separate apps to Codemagic so I can capture the App IDs, one from your current app (linked to the old repository) and one from the newly added app (linked to the new repository).

Here’s what I need from you:
﻿​﻿
﻿1. The App ID of the application whose URL needs to be changed (the one currently linked to the old repository)
- You can find this App ID in the URL when viewing the application in Codemagic.

2. The App ID of a newly added application - Add a new application to Codemagic - the one that is now in a different location (the new repo) and share the app ID with me

I need both App IDs in Codemagic in order to process the update.

﻿Please provide the details in this format:
Old App ID:
New App ID:
​﻿
Once I update the repository URL of your existing application, all environment variables, build history, and settings will remain intact, only the repository URL will change. After the update, you’ll see that the original app now points to the new repository. At this point, you can safely remove the newly added application if you wish, since it will no longer be needed.

**Reply by dtrdic** [reply]
Hi @Adeiza-Dev I'm tentatively closing the thread since we haven't heard back from you. You can open the thread again by replying to this message

**Reply by Adeiza-Dev** [reply]
Hi, Thank you @dtrdic . I started having this issue again. I will do the steps you listed

**Reply by Adeiza-Dev** [reply]
I am unable to create the new app with the right repo. Because even though it is connected to codemagic. it doesn't show up when I want to create a new application. Please helpppppp

**Reply by dtrdic** [reply]
Hi @Adeiza-Dev, that usually means this repository has already been added to your team, or another team. Codemagic doesn't allow the same repository to be added multiple times using OAuth authentication.

If you need to add the same app again, you can do so by importing it using the repository URL instead.

Please follow these steps for adding your repository
​﻿
﻿1. Click on Add application
2. Select team and connect repo using the "Other" option
<img width="747" height="499" alt="image" src="https://github.com/user-attachments/assets/f687af51-2c7b-4f01-854a-203f6c47ff81" />

3. Clone your repo using SSH option

<img width="418" height="339" alt="image" src="https://github.com/user-attachments/assets/d5b1a3a7-c034-48cd-8280-877f414f5994" />


4. Click on generate key button and copy

<img width="726" height="986" alt="image" src="https://github.com/user-attachments/assets/ca223597-d01d-408b-be0a-f30353645a1d" />

5. Go to https://github.com/settings/keys and add new SSH key

**Reply by Adeiza-Dev** [reply]
Thank you. trying this now

**Reply by Adeiza-Dev** [reply]
New app id: 6a8555cf47a242f14993cbf8
Old app id: 641c18064aa7a138692ff6a6

Thank you. @dtrdic

**Reply by Adeiza-Dev** [reply]
@dtrdic ...

**Reply by dtrdic** [reply]
@Adeiza-Dev Done

**Reply by Adeiza-Dev** [reply]
Worked. Thank youuu

**Reply by Adeiza-Dev** [reply]
I don't know I think something weird happened. 
Just noticed that it happened to both apps on my team.

I did the same step for the other project

New App id: 6a85627f1ac2da2d65615046
Old App id: 6921dff40f9548d1bbb9e38a
Thanks @dtrdic  David

**Reply by Adeiza-Dev** [reply]
@dtrdic In case you missed this

**Reply by dtrdic** [reply]
@Adeiza-Dev done.

**Reply by Adeiza-Dev** [reply]
Thank you so much

**Comment by Brackyt** [comment]
Hi, it's still not possible to update the repo in Codemagic. Does it still work if I ask here ? @dtrdic 
Old App Id: 691a0f69506ead9eb6bae1a4
New App Id: 6a21ea26363a031ec8f5fc95

Thanks if you can make the migration

**Reply by dtrdic** [reply]
Hi @Brackyt, my apologies i only saw your message now..
Done. Your old repo url is now updated in Codemagic
